istioctl/cmd/auth_test.go

```diff
@@ -23,30 +23,56 @@ import (
 	"istio.io/istio/pilot/test/util"
 )
 
+func runCommandAndCheckGolden(name, command, golden string, t *testing.T) {
+	t.Helper()
+	var out bytes.Buffer
+	rootCmd := GetRootCmd(strings.Split(command, " "))
+	rootCmd.SetOutput(&out)
+
+	err := rootCmd.Execute()
+	if err != nil {
+		t.Errorf("%s: unexpected error: %s", name, err)
+	} else {
+		util.CompareContent(out.Bytes(), golden, t)
+	}
+}
+
 func TestAuthCheck(t *testing.T) {
 	testCases := []struct {
-		name           string
-		in             string
-		goldenFilename string
+		name   string
+		in     string
+		golden string
 	}{
 		{
-			name:           "listeners and clusters",
-			in:             "testdata/auth/productpage_config_dump.json",
-			goldenFilename: "testdata/auth/productpage.golden"},
+			name:   "listeners and clusters",
+			in:     "testdata/auth/productpage_config_dump.json",
+			golden: "testdata/auth/productpage.golden",
+		},
 	}
 
 	for _, c := range testCases {
-		var out bytes.Buffer
-
 		command := fmt.Sprintf("experimental auth check -f %s", c.in)
-		rootCmd := GetRootCmd(strings.Split(command, " "))
-		rootCmd.SetOutput(&out)
-
-		err := rootCmd.Execute()
-		if err != nil {
-			t.Errorf("%s: unexpected error: %s", c.name, err)
-		} else {
-			util.CompareContent(out.Bytes(), c.goldenFilename, t)
-		}
+		runCommandAndCheckGolden(c.name, command, c.golden, t)
+	}
+}
+
+func TestAuthUpgrade(t *testing.T) {
```

Comment:
Maybe change the name to TestAuthzUpgrade or TestRbacUpgrade? Later if we want to test AuthN upgrade, we can just create another function to decouple the tests?

Reply:
I would like to keep this name, in the future, we can have both authN and authZ test cases here, they can even appear in the same input file. From the end user, they only need to invoke the `upgrade` command and it works for both authN and authZ.